### PR TITLE
pythonPackages.uamqp: Fix build on darwin

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/pull/109473

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/uamqp/default.nix [hunk 1]
```diff
@@ -18,14 +22,25 @@ buildPythonPackage rec {
     sha256 = "sha256-zDUFe/yMCThn+qJqDekMrUHEf1glGxBw4pioExLLoqg=";
   };
 
+  # can be removed when https://github.com/Azure/azure-c-shared-utility/pull/499 is in the vendored files
```
Comment (Mic92) [inline pkgs/development/python-modules/uamqp/default.nix:25]: fetchpatch does not work here?

Comment (SuperSandro2000) [reply pkgs/development/python-modules/uamqp/default.nix:25]: I did the PR afterwards. The files are vendored and the last time I fiddled with patching vendors it took me way to long to get it working. I guess we could do it here but I am lazy right now.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/uamqp/default.nix:25]: Guess I got better and fixed it right away.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109538` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.azure-servicebus</li>
</ul>
</details>
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>python37Packages.azure-servicebus</li>
<li>python37Packages.uamqp</li>
<li>python38Packages.azure-servicebus</li>
<li>python38Packages.uamqp</li>
<li>python39Packages.uamqp</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 109538` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python37Packages.azure-servicebus</li>
    <li>python37Packages.uamqp</li>
    <li>python38Packages.azure-servicebus</li>
    <li>python38Packages.uamqp</li>
    <li>python39Packages.uamqp</li>
  </ul>
</details>

**Comment by jonringer** [comment]
wow, and a patch, thanks @SuperSandro2000 wasn't expecting it to be this much work. But also shame on microsoft on not have CI around darwin.

**Pull request merged**